src/requirements/requirement-frontend-utils.ts

```diff
@@ -208,32 +244,25 @@ export function getMatchedRequirementFulfillmentSpecification(
   }
 }
 
-/**
- * @param requirement the requirement which we want sub-requirement progress report.
- * @param coursesTaken a list of courses that are associated with the requirement in the graph.
- * @param toggleableRequirementChoices user's choices on toggleable requirements.
- * @returns requirement fulfillment stat, and partitioned courses into sub-requirement slots.
- */
-export function computeFulfillmentCoursesAndStatistics(
-  requirement: RequirementWithIDSourceType,
-  coursesTaken: readonly CourseTaken[],
-  toggleableRequirementChoices: AppToggleableRequirementChoices
-): RequirementFulfillmentStatistics & { readonly courses: readonly (readonly CourseTaken[])[] } {
-  const spec = getMatchedRequirementFulfillmentSpecification(
-    requirement,
-    toggleableRequirementChoices
-  );
-  if (spec == null) {
-    // Give self-check 1 required course and 0 fulfilled to prevent it from being fulfilled.
-    return { fulfilledBy: 'self-check', minCountFulfilled: 0, minCountRequired: 1, courses: [] };
-  }
-  const { fulfilledBy, eligibleCourses, perSlotMinCount, minNumberOfSlots } = spec;
+export type RequirementFulfillmentStatisticsWithCourses = RequirementFulfillmentStatistics & {
+  readonly courses: readonly (readonly CourseTaken[])[];
+};
 
+const computeFilfillmentStatistics = (
```

Comment:
I think it should be `computeFulfillmentStatistics`

Reply:
fixed